src/timezones/types.jl

```diff
@@ -0,0 +1,212 @@
+
+# import Base.Dates: UTInstant, DateTime, TimeZone, Millisecond
+using Base.Dates
+import Base.Dates: value
+
+# Note: The Olsen Database rounds offset precision to the nearest second
+# See "America/New_York" notes for an example.
+
+# Using type Symbol instead of AbstractString for name since it
+# gets us ==, and hash for free.
+immutable FixedTimeZone <: TimeZone
+    name::Symbol
+    utc_offset::Second  # Standard offset from UTC
+    dst_offset::Second  # Addition offset applied to UTC offset
+
+    function FixedTimeZone(name::Symbol, utc_offset::Second, dst_offset::Second=Second(0))
+        new(name, utc_offset, dst_offset)
+    end
+end
+
+function FixedTimeZone(name::String, utc_offset::Int, dst_offset::Int=0)
+    FixedTimeZone(symbol(name), Second(utc_offset), Second(dst_offset))
```

Comment:
Do we want to validate that `name` is a valid TimeZone? Or is the point that you can create custom timezone types?